numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -78,6 +78,29 @@ PyArray_OutputConverter(PyObject *object, PyArrayObject **address)
     }
 }
 
+static npy_intp
+intp_from_scalar(PyObject *ob)
```

Comment:
After thinking about, it maybe we can call it `dimension_from_scalar`, makes it a bit clearer where it is used (normally?) for, and why the error is: Maximum allowed dimension exceeded.

Reply:
Fixed in fd1c1ceb6924cce4122bee14fe3b20f2f3ec2d8a